core/util/runPuppet.js

```diff
@@ -421,8 +421,18 @@ async function captureScreenshot (page, browser, selector, selectorMap, config,
       if (el) {
         const box = await el.boundingBox();
         if (box) {
+          // Resize the viewport to screenshot elements outside of the viewport
```

Comment:
@kasparsz   Thank you for this thoughtful solution.  Would you mind adding a feature flag here? Maybe call it `useBoundingBoxViewportForSelectors` we can default to "on" e.g. 
```
if (config.useBoundingBoxViewportForSelectors === false) {
    Your resize fix...
}
```
This way -- if someone has an issue we can advise them to disable this feature.

Reply:
You'll need to handle for `undefined` etc.   Please add tests for this also.

Thank you!

Reply:
Thank You

Commit: https://github.com/kasparsz/BackstopJS/commit/e8fda8ded539a814d0bf730deb0a9fab6d112bce

```if (config.useBoundingBoxViewportForSelectors !== false) { ... fix ... }```
Added, but since it's on by default, anything except for "false" would enable it, so"undefined" would still mean it's on, so "!== false" should be sufficient in that case. Or did I misunderstood you?

Test puppet_backstop_features_hideSelectors_0_document_0_phone.png before this fix was failing already consistently, a test for this would be a copy of it.